crates/uv/src/commands/project/run.rs

```diff
@@ -451,6 +451,12 @@ pub(crate) async fn run(
                 .await?
             };
 
+            if project.workspace().pyproject_toml().is_packaged_script()
```

Comment:
Note: sync is performed on CLI add/remove/sync/run, but it seems to annoying to warn on each command. Thus this warning is only displayed at `uv run` time, which should be the most effective place. But can be expanded if we prefer.

Reply:
I think it makes sense to omit at `add` / `remove`. We might want it at `sync`, otherwise if you sync then active the environment (instead of using `uv run`), you'll never see it.

Maybe with some special-casing per https://github.com/astral-sh/uv/pull/7420/files#r1761054908 we can make the warnings quieter in `uv run`.